master/pkg/searcher/search_method.go

```diff
@@ -15,30 +15,30 @@ type context struct {
 // interface should use pointer receivers to ensure interface equality is calculated through pointer
 // equality.
 type SearchMethod interface {
-	// initialOperations returns a set of initial operations that the searcher would like to take.
+	// initialRuns returns a set of initial runs the searcher would like to create.
 	// This should be called only once after the searcher has been created.
-	initialOperations(ctx context) ([]Operation, error)
-	// trialCreated informs the searcher that a trial has been created as a result of a Create
+	initialRuns(ctx context) ([]Action, error)
+	// runCreated informs the searcher that a run has been created as a result of a Create
+	// action and returns additional Actions to perform.
+	runCreated(ctx context, runID int32, action Create) ([]Action, error)
+	// validationCompleted informs the searcher that a validation metric has been reported.
+	// xxx: reword comments
+	validationCompleted(ctx context, runID int32,
+		metrics map[string]interface{}) ([]Action, error)
+	// runClosed informs the searcher that the trial has been closed as a result of a Close
 	// operation.
-	trialCreated(ctx context, requestID model.RequestID) ([]Operation, error)
-	// validationCompleted informs the searcher that the validation workload initiated by the same
-	// searcher has completed. It returns any new operations as a result of this workload
-	// completing.
-	validationCompleted(ctx context, requestID model.RequestID,
-		metric interface{}, op ValidateAfter) ([]Operation, error)
-	// trialClosed informs the searcher that the trial has been closed as a result of a Close
-	// operation.
-	trialClosed(ctx context, requestID model.RequestID) ([]Operation, error)
-	// progress returns experiment progress as a float between 0.0 and 1.0.
-	progress(map[model.RequestID]PartialUnits, map[model.RequestID]bool) float64
-	// trialExitedEarly informs the searcher that the trial has exited earlier than expected.
-	trialExitedEarly(
-		ctx context, requestID model.RequestID, exitedReason model.ExitedReason,
-	) ([]Operation, error)
+	runClosed(ctx context, runID int32) ([]Action, error)
+	// progress returns search progress as a float between 0.0 and 1.0.
+	progress(map[int32]float64, map[int32]bool) float64
+	// runExitedEarly informs the searcher that the run has exited earlier than expected.
+	runExitedEarly(
+		ctx context, runID int32, exitedReason model.ExitedReason,
+	) ([]Action, error)
 
 	// TODO: refactor as model.Snapshotter interface or something
 	model.Snapshotter
 	expconf.InUnits
```

Comment:
shouldn't we be able to get rid of exconf.InUnits as part of this refactor?

Reply:
actually, I mean that as in "we should remove it from this interface" but I suppose it's actually just pointless now, and can be deleted from the entire codebase, right?

Reply:
yes and yes and done